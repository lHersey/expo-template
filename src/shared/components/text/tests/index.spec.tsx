import { render } from '@testing-library/react-native';
import React from 'react';
import Providers from 'shared/providers';

import Text from '../index';

describe('<Text />', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
      expect.hasAssertions();
      const rendered = render(
        <Providers>
          <Text text="Hello Text" />
        </Providers>,
      ).toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('unit', () => {
    it('should render the text by prop', () => {
      expect.hasAssertions();

      const text = 'Hello Text';
      const { queryByText } = render(
        <Providers>
          <Text text={text} />
        </Providers>,
      );

      expect(queryByText(text)).toBeTruthy();
    });

    it('should render the text by children', () => {
      expect.hasAssertions();

      const text = 'Hello Text';
      const { queryByText } = render(
        <Providers>
          <Text>{text}</Text>
        </Providers>,
      );

      expect(queryByText(text)).toBeTruthy();
    });
  });
});
