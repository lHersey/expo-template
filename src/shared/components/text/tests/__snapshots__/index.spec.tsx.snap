// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> snapshots should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#102A43",
      },
    ]
  }
>
  Hello Text
</Text>
`;
